src/components/accordion/accordion.test.js

```diff
@@ -0,0 +1,183 @@
+      <div>Content</div>
+    </Accordion>
+  );
+};
+
+const AccordionWithIcon = () => {
+  const [errors] = React.useState({
+    one: "error",
+  });
+
+  const [expanded, setExpanded] = React.useState({
+    one: false,
+  });
+
+  return (
+    <AccordionGroup>
+      <Accordion
+        title="Heading"
+        expanded={expanded.one}
+        onChange={() =>
+          setExpanded((previousState) => ({
+            ...previousState,
+            one: !previousState.one,
+          }))
+        }
+        error={errors.one}
+      >
+        <Checkbox label="Add error" />
+      </Accordion>
+    </AccordionGroup>
+  );
+};
+
+context("Testing Accordion component", () => {
+  describe("should render Accordion component", () => {
+    it("should check AccordionRow is expanded using click", () => {
+      CypressMountWithProviders(<AccordionComponent />);
+
+      accordionTitleContainer().click();
+
+      accordionTitleContainer()
+        .should("have.attr", "aria-expanded", "true")
+        .and("be.visible");
+    });
+
+    it("should check AccordionRow is expanded using Enter key", () => {
+      CypressMountWithProviders(<AccordionComponent />);
+
+      accordionTitleContainer().trigger("keydown", keyCode("Enter"));
+
+      accordionTitleContainer()
+        .should("have.attr", "aria-expanded", "true")
+        .and("be.visible");
+    });
+
+    it("should verify AccordionRow has golden border outline", () => {
+      CypressMountWithProviders(<AccordionComponent />);
+
+      accordionTitleContainer().click();
+
+      accordionDefaultTitle()
+        .should("have.css", "outline", "rgb(255, 181, 0) solid 3px")
+        .and("be.visible");
+    });
+
+    it.each([["chevron_down"], ["dropdown"]])(
+      "when Accordion row is closed should set iconType to %s",
+      (iconType) => {
+        CypressMountWithProviders(<AccordionComponent iconType={iconType} />);
+
+        accordionIcon()
+          .should("have.attr", "type", iconType)
+          .and("be.visible")
+          .and(
+            "have.css",
+            "transform",
+            "matrix(6.12323e-17, 1, -1, 6.12323e-17, 0, 0)"
+          );
+      }
+    );
+
+    it.each([["chevron_down"], ["dropdown"]])(
+      "when Accordion row is open should set iconType to %s",
+      (iconType) => {
+        CypressMountWithProviders(
+          <AccordionComponent iconType={iconType}> </AccordionComponent>
+        );
+
+        accordionTitleContainer().click();
+
+        accordionIcon()
+          .should("have.attr", "type", iconType)
+          .and("be.visible")
+          .and(
+            "not.have.css",
+            "transform",
+            "matrix(6.12323e-17, 1, -1, 6.12323e-17, 0, 0)"
+          );
+      }
+    );
+
+    it.each([["left"], ["right"]])(
+      "should set Accordion iconAlign to %s",
+      (iconAlign) => {
+        CypressMountWithProviders(
+          <AccordionComponent iconAlign={iconAlign}> </AccordionComponent>
```

Comment:
We do not need to add a closure if we are not passing the `children` to component
```suggestion
          <AccordionComponent iconAlign={iconAlign} />
```